### PR TITLE
fix most flake8 warnings

### PR DESCRIPTION
**Pull request by Esgrove**
One more PR with some small formatting and syntax fixes 😅 

* Fix most flake8 warnings, only ones remaining are function complexity warnings
* Turns out ruff only supports 3.7+ so updated the config just for it

### CODE REVIEW

File: n_utils/tests/profile_util_test.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ def test_adfs_profile_switch(capsys):
     assert "adfs-aws-login --profile testprofile" in captured.out
 
 
-def test_adfs_profile_switch(capsys):
+def test_lastpass_profile_switch(capsys):
     enable_profile("lastpass", "testprofile")
     captured = capsys.readouterr()
-    assert "lastpass-aws-login --profile testprofile"
+    assert "lastpass-aws-login --profile testprofile" in captured.out
```
Comment (Esgrove) [inline n_utils/tests/profile_util_test.py:37]: This test was not doing anything and the name conflicted with the one above 🤔 Is this needed or should it be removed?

File: n_utils/aws_infra_util.py [hunk 2]
```diff
@@ -410,9 +409,10 @@ def resolve_ami(component_params, component, image, imagebranch, branch, git):
             # resolve with a specifically set image build number
             build = component_params[build_param]
             image_tag = job + "_" + build
-            job_tag_func = (
-                lambda image, image_name_prefix: len([tag for tag in image["Tags"] if tag["Value"] == image_tag]) > 0
-            )
+
+            def job_tag_func(image, image_name_prefix):
+                return len([tag for tag in image["Tags"] if tag["Value"] == image_tag]) > 0
```
Comment (Esgrove) [inline n_utils/aws_infra_util.py:414]: https://www.flake8rules.com/rules/E731.html

File: n_utils/__init__.py [hunk 3]
```diff
@@ -150,7 +150,7 @@ def _to_str(data):
     if callable(decode_method):
         try:
             ret = data.decode()
-        except:
+        except Exception:
```
Comment (Esgrove) [inline n_utils/__init__.py:153]: https://www.flake8rules.com/rules/E722.html

File: n_utils/tests/profile_util_test.py (hunk 1 above)
Comment (psiniemi) [reply n_utils/tests/profile_util_test.py:37]: Not a huge test, but supposed to test enabling lastpass saml authenticated profiles. I guess that's a cut & paste bug (incomplete and forgot to change function name).

**Pull request merged**